python/lsst/meas/astrom/anetAstrometry.py

```diff
@@ -0,0 +1,464 @@
+                                with distorted points in "centroid.distorted" field.
+        \return bounding box of distorted exposure
+        """
+        detector = exposure.getDetector()
+        pixToTanXYTransform = None
+        if detector is None:
+            self.log.warn("No detector associated with exposure; assuming null distortion")
+        else:
+            tanSys = detector.makeCameraSys(TAN_PIXELS)
+            pixToTanXYTransform = detector.getTransformMap().get(tanSys)
+
+        if pixToTanXYTransform is None:
+            self.log.info("Null distortion correction")
+            for s in sourceCat:
+                s.set(self.centroidKey, s.getCentroid())
+                s.set(self.centroidErrKey, s.getCentroidErr())
+                s.set(self.centroidFlagKey, s.getCentroidFlag())
+            return exposure.getBBox()
+
+        # Distort source positions
+        self.log.info("Applying distortion correction")
+        for s in sourceCat:
+            centroid = pixToTanXYTransform.forwardTransform(s.getCentroid())
+            s.set(self.centroidKey, centroid)
+            s.set(self.centroidErrKey, s.getCentroidErr())
+            s.set(self.centroidFlagKey, s.getCentroidFlag())
+
+
+        # Get distorted image size so that astrometry_net does not clip.
+        bboxD = afwGeom.Box2D()
+        for corner in detector.getCorners(TAN_PIXELS):
+            bboxD.include(corner)
+        return afwGeom.Box2I(bboxD)
+
+    @contextmanager
+    def distortionContext(self, sourceCat, exposure):
+        """!Context manager that applies and removes distortion
+
+        We move the "centroid" definition in the catalog table to
+        point to the distorted positions.  This is undone on exit
+        from the context.
+
+        The input Wcs is taken to refer to the coordinate system
+        with the distortion correction applied, and hence no shift
+        is required when the sources are distorted.  However, after
+        Wcs fitting, the Wcs is in the distorted frame so when the
+        distortion correction is removed, the Wcs needs to be
+        shifted to compensate.
+
+        \param exposure Exposure holding Wcs
+        \param sourceCat Sources to correct for distortion
+        \return bounding box of distorted exposure
+        """
+        # Apply distortion
+        bbox = self.distort(sourceCat=sourceCat, exposure=exposure)
+        oldCentroidName = sourceCat.table.getCentroidDefinition()
+        sourceCat.table.defineCentroid(self.distortedName)
+        try:
+            yield bbox # Execute 'with' block, providing bbox to 'as' variable
+        finally:
+            # Un-apply distortion
+            sourceCat.table.defineCentroid(oldCentroidName)
+            x0, y0 = exposure.getXY0()
+            wcs = exposure.getWcs()
+            if wcs:
+                wcs.shiftReferencePixel(-bbox.getMinX() + x0, -bbox.getMinY() + y0)
```

Comment:
I'm afraid I didn't take the time to try to figure out the complications involved in distorting and undistorting pixels. This is old code that I only cosmetically touched. I'm hoping it will entirely go away, but if it sticks around it desperately needs an overhaul.